### PR TITLE
Fix possible missing class error

### PR DESCRIPTION
**Pull request by alpar-t**
- Use the `testRuntime` classpath when loading the tests instead of
compile
- simplify task action by moving dealing with reandomized testing task vs gradle in the input method
- improve error message when classes are missing

### CODE REVIEW

File: buildSrc/src/main/java/org/elasticsearch/gradle/precommit/TestingConventionsTasks.java [hunk 1]
```diff
@@ -295,7 +288,10 @@ private boolean seemsLikeATest(Class<?> clazz) {
             }
             return false;
         } catch (NoClassDefFoundError e) {
-            throw new IllegalStateException("Failed to inspect class " + clazz.getName(), e);
+            // Include the message to get more info to get more a more useful message when running Gradle without -s
+            throw new IllegalStateException(
+                "Failed to inspect class " + clazz.getName() + ". Missing class? " + e.getMessage(),
```
Comment (rjernst) [inline buildSrc/src/main/java/org/elasticsearch/gradle/precommit/TestingConventionsTasks.java]: The cause is already attached, is duplicating the cause's message really necessary?

File: buildSrc/src/main/java/org/elasticsearch/gradle/precommit/TestingConventionsTasks.java [hunk 2]
```diff
@@ -323,9 +319,11 @@ private boolean isAnnotated(Class<?> clazz, Method method, Class<?> annotation)
     }
 
     private FileCollection getTestsClassPath() {
-        // This is doesn't need to be annotated with @Classpath because we only really care about the test source set
+        // Loading the classes does depend on the @Classpath but we can have better caching if we don't annotate this
```
Comment (rjernst) [inline buildSrc/src/main/java/org/elasticsearch/gradle/precommit/TestingConventionsTasks.java]: nit: What is the `@` for with classpath? It makes it look like an annotation, but I don't think it is?

File: buildSrc/src/main/java/org/elasticsearch/gradle/precommit/TestingConventionsTasks.java [hunk 3]
```diff
@@ -323,9 +319,11 @@ private boolean isAnnotated(Class<?> clazz, Method method, Class<?> annotation)
...
+        // as such as the conventions as such don't depend on the classpath and any missing classes would be detected
```
Comment (rjernst) [inline buildSrc/src/main/java/org/elasticsearch/gradle/precommit/TestingConventionsTasks.java]: This sentence doesn't make grammatical sense...I can't figure out what was intended either.

**Review by rjernst — APPROVED** [review]
Looks fine. Left a few nits.

File: buildSrc/src/main/java/org/elasticsearch/gradle/precommit/TestingConventionsTasks.java (hunk 1 above)
Comment (atorok) [reply buildSrc/src/main/java/org/elasticsearch/gradle/precommit/TestingConventionsTasks.java]: I did it so we get the name of the missing class even when the build does not run with  `-s`. Would be useful in CI.

**Pull request merged**